test/tools/utils.js

```diff
@@ -0,0 +1,236 @@
+    if (srcDataDict['inputsData'] !== undefined &&
+        srcDataDict['inputsData'][source] !== undefined) {
+      dstDataDict['inputsData'][source] = srcDataDict['inputsData'][source];
+    } else {
+      const targetDataInfo = inputsDataInfo[source];
+      if (targetDataInfo.data !== undefined) {
+        const srcDataInfo = inputsDataInfo[targetDataInfo.data];
+        const inputTensor = new Tensor(
+            srcDataInfo.shape, dstDataDict['inputsData'][targetDataInfo.data]);
+        let outputTensor;
+        if (targetDataInfo.processCategory === 'transpose') {
+          outputTensor = transpose(
+              inputTensor, {permutation: targetDataInfo.permutation});
+        } else if (targetDataInfo.processCategory === 'negative') {
+          outputTensor = neg(inputTensor);
+        }
+        console.log(`source ${source}`);
+        dstDataDict['inputsData'][source] = outputTensor.data;
+      } else {
+        const total = sizeOfShape(targetDataInfo.shape);
+        const sign = targetDataInfo.sign;
+        const type = targetDataInfo.type;
+        if (targetDataInfo.dataRange) {
+          min = targetDataInfo.dataRange[0];
+          max = targetDataInfo.dataRange[1];
+        }
+        const generatedNumbers =
+            utils.getRandomNumbers(min, max, total, type, sign);
+        dstDataDict['inputsData'][source] = generatedNumbers;
+      }
+    }
+  }
+  return dstDataDict;
+}
+
+/**
+ * Get JSON object from specified JSON file.
+ * @param {String} filePath
+ * @return {Object}
+ */
+function readJsonFile(filePath) {
+  let inputFile;
+  if (path.isAbsolute(filePath)) {
+    inputFile = filePath;
+  } else {
+    inputFile =
+        path.join(path.dirname(process.argv[1]), filePath);
+  }
+  const content = fs.readFileSync(inputFile).toString();
+  const jsonDict = JSON.parse(
+      content.replace(/\\"|"(?:\\"|[^"])*"|(\/\/.*|\/\*[\s\S]*?\*\/)/g,
+          (m, g) => g ? '' : m));
+  return jsonDict;
+}
+
+/**
+ * Save JSON infomation into file.
+ * @param {Object} jsonDict
+ * @param {String} saveFile
+ */
+function writeJsonFile(jsonDict, saveFile) {
+  const parentDirectory = path.dirname(saveFile);
+  if (!fs.existsSync(parentDirectory)) {
+    fs.mkdirSync(parentDirectory);
+  }
+  const jsonString = JSON.stringify(jsonDict, function(key, value) {
+    // the replacer function is looking for some typed arrays.
+    // If found, it replaces it by a trio
+    if ( value instanceof Int8Array ||
+         value instanceof Uint8Array ||
+         value instanceof Uint16Array ||
+         value instanceof Int32Array ||
+         value instanceof Uint32Array ||
+         value instanceof Float32Array ||
+         value instanceof Float16Array ||
+         value instanceof Float64Array ) {
+      if (value.length ===1) { // value instanceof Uint8Array &&
+        const result = [];
+        result[0] = value[0];
+        return result;
+      } else {
```

Comment:
Is this the `BigInt64Array` case? If so, comment is nice.

Reply:
No, this `else` block is for the array which length is greater than **1**. For the array having only one value, `Array.apply([], array)` doesn't work, we need specially handle it as above block. Thanks!